agot-bg-game-server/src/client/game-state-panel/ResolveSingleMarchOrderComponent.tsx

```diff
@@ -127,6 +112,24 @@ export default class ResolveSingleMarchOrderComponent extends Component<GameStat
         );
     }
 
+    renderTooltip(reason: string) {
+        return <Tooltip id={"leave-power-token"}>
+            {reason == "already-capital" ? (
+                "Your capital is always controlled by your house, thus not requiring a Power"
+                + " token to be left when leaving the area to keep control of it."
+            ) : reason == "already-power-token" ? (
+                "A Power token is already present."
+            ) : reason == "no-power-token-available" ? (
+                "You don't have any available Power token."
+            ) : reason == "not-a-land" ? (
+                "Power tokens can only be left on land areas."
+            ) : reason == "no-all-units-go" ? (
+                "All units must leave the area in order to leave a Power token."
+            ) : "Leaving a Power token in an area maintain the control your house has on it, even"
+                    + " if all units your units leave the area."}
+        </Tooltip>
+    }
+
```

Comment:
Was there a special reason for this change?

I'm not a fan of doing this in React, as it requires the reader to jump to multiple part of the files to understand the code of the UI. I agree that I tend to be lazy when it comes to splitting things in Components, which leads to huge files with a big number of indentation.

Reply:
Ouh well.. My first approach made the label clickable but the tooltips were gone. So I played around and debugged why. Finally there was no need for this but I decided to keep it because I am a fan of less complex methods by splitting huge methods in several small ones. As modern IDEs support Jump to definition I don’t see that as a problem. But of course it’s no problem to change it back.

Reply:
Fixed